### ISSUE TITLE
[windows] import.meta.url causes Segmentation Fault when import contains a query parameters

### ISSUE THREAD

**Issue by ftoh** [oven-sh/bun#8538]
### What version of Bun is running?

~1.0.26-canary.79+5a0ae3c25~
~1.0.26-canary.4+1f0c1c81e~
~1.0.26-canary.63+d0b6218bd~
~1.0.33-canary.10+931e04b01~

**UPD. Fixed in 1.0.36-canary.40+6164fac25**

### What platform is your computer?

Microsoft Windows NT 10.0.19044.0 x64

### What steps can reproduce the bug?

1. Create file **main.ts**
```ts
import './test?param=value'
```
2. Create file **test.ts**
```ts
console.log('file test.ts', import.meta.url)
```
3. Run `bun run main.ts` multiple times

### What is the expected behavior?

```sh
PS D:\app> bun run main.ts
file test.ts file:///D:/app/test.ts%3Fparam=value
```

### What do you see instead?

Result is not stable, sometimes it works fine
```sh
PS D:\app> bun run main.ts
file test.ts file:///D:/app/test.ts%3Fparam=value
PS D:\app> bun run main.ts
file test.ts file:///D:/app/test.ts%3Fparam=value
PS D:\app> bun run main.ts
Segmentation fault at address 0x2
???:?:?: 0x7ff7e79650e4 in ??? (bun.exe)
???:?:?: 0x7ff7e7964d65 in ??? (bun.exe)
???:?:?: 0x7ff7e791daed in ??? (bun.exe)
???:?:?: 0x7ff7e791e513 in ??? (bun.exe)
???:?:?: 0x7ff7e791c916 in ??? (bun.exe)
???:?:?: 0x1858000115d in ??? (???)
???:?:?: 0x1858000111f in ??? (???)
PS D:\app> bun run main.ts
file test.ts 荸Ȅ胸Ȅ⼃⼯㩃啜愐Ȅ䑜獥瑫䍐Ȅ嬐Ȅ@혊쀣ༀ退ȄȄȄ
PS D:\app> bun run main.ts
Segmentation fault at address 0x8ffe993c4
???:?:?: 0x7ff7e9001f89 in ??? (bun.exe)
???:?:?: 0x7ff7e798116a in ??? (bun.exe)
???:?:?: 0x7ff7e81df51c in ??? (bun.exe)
???:?:?: 0x7ff7e81de991 in ??? (bun.exe)
???:?:?: 0x7ff7e909b77c in ??? (bun.exe)
???:?:?: 0x174fec8af27 in ??? (???)
???:?:?: 0x7ff7e77db2a1 in ??? (bun.exe)
???:?:?: 0x174fec94127 in ??? (???)
```

### Additional information

This code works fine in WSL

**Comment by paperclover** [comment]
it actually is just a module with a question mark like this

**Comment by ftoh** [comment]
@Jarred-Sumner I have checked a latest release, it still failed.
```pwsh
❯ bun --revision
1.0.33-canary.10+931e04b01
❯ bun run-test.ts -s -r 30 bun -e "import\(\'./test.ts?param=value\'\)"
Command: bun -e import\(\'./test.ts?param=value\'\)
Number of fails: 10
Number of successes: 20
```

<details>
<summary>run-test.ts</summary>

```js
/**
 * This script is used to repeat a command multiple times and/or in silent mode.
 *
 * @description This script allows users to run a command multiple times and/or in silent mode. The script accepts command line arguments to specify the number of times to repeat the command and whether to run in silent mode.
 *
 * @param {-s} Specifies that the script should run in silent mode.
 * @param {-r} Specifies the number of times to repeat the command.
 *
 * @example
 * bun run-test.ts -s -r 10 "dir"
 * This command runs the "dir" command 10 times in silent mode.
 *
 */
import { $ } from 'bun'

let silent = false
let repeat = 1

let args = process.argv.slice(process.argv.indexOf(__filename) + 1)

if (args[0] === '-s') {
  silent = true
  args = args.slice(1)
}

if (args[0] === '-r') {
  repeat = parseInt(args[1])
  args = args.slice(2)
}

const cmd = args.join(' ')

console.log(`Command: ${cmd}`)

let numFails = 0,
  numSuccesses = 0
for (let i = 0; i < repeat; i++) {
  try {
    let p = $`${{ raw: cmd }}`.throws(true)
    if (silent) {
      p.quiet()
    }
    await p

    numSuccesses += 1
  } catch (error) {
    numFails += 1
  }
}

console.log(`Number of fails: ${numFails}`)
console.log(`Number of successes: ${numSuccesses}`)

```
</details>


**UPD. Fixed in 1.0.36-canary.40+6164fac25**
